### DISCUSSION TITLE
3차 데모데이 전 QA

### DISCUSSION THREAD

**Discussion by kyum-q** [woowacourse-teams/2024-code-zap#345] [QA]
## 3차 데모데이 전 QA

**Comment by kyum-q** [comment]
- 카테고리 없음 정렬 맨 밑으로
- 카테고리 검색 및 목록 조회 시 keyword 2개 가는거 제거

**Reply by zangsu** [reply]
keyword 2개 가는 거 해결 완료 - 마위

**Reply by kyum-q** [reply]
백에서 정렬해서 보내주기로 
- 카테고리 없음이 맨위로 가도록 (ID asc)

**Comment by zangsu** [comment]
새로 회원가입을 한 경우 "카테고리 없음" 카테고리가 생성되지 않음

**Reply by zangsu** [reply]
-> 로그인이 실패한 경우에도 로그인 성공으로 처리가 되어서 실제로 존재하지 않는 유저의 창이 보였던 문제였음
로그인 실패 처리를 하여 해결 완료 - 마위

**Comment by Hain-tain** [comment]
- 템플릿 생성 이후 새로 생성된 템플릿으로 이동해야하는데, 전체 목록 페이지로 이동함

<정상 동작>
템플릿 생성 페이지에서 저장 => 새로 생성된 템플릿 페이지로 이동

<현재 동작>
템플릿 생성 페이지에서 저장 =>  전체 목록 페이지로 이동

=> 현재 헤더에서 로케이션 정보 따와서 이동시켜놨는데 이 부분이 왜인지 모르지만 정상적으로 동작하고있지 않음

**Comment by zangsu** [comment]
로그인을 한 이후에도 새 템플릿 작성 창으로 이동하면 인증 정보가 사라짐
-> 카테고리 목록을 조회하려고 하는데 401 이 반환됨

**Reply by zangsu** [reply]
실제 DB에 있는 로그인 정보가 맞는지 다시 확인을 해 봐야 할 것 같음

**Reply by zangsu** [reply]
-> 로그인이 실패한 경우에도 로그인 성공으로 처리가 되어서 실제로 존재하지 않는 유저의 창이 보였던 문제였음
로그인 실패 처리를 하여 해결 완료 - 마위

**Comment by vi-wolhwa** [comment]
### 카테고리 요청
> 카테고리 요청에 대한 응답이 정렬되지 않음

**Comment by zangsu** [comment]
카테고리를 아무 것도 선택하지 않고 ("카테고리 없음" 도 선택하지 않고) 템플릿을 생성하니 401 이 발생

**Reply by zangsu** [reply]
FE -> "카테고리 없음" 이 무조건 1번이 아니라서 해당 유저의 "카테고리 없음" 을 선택해 주어야 함
BE -> 카테고리에 대한 권한이 없는 경우는 `403 FORBIDDEN` 을 사용하도록 변경해 주어야 함

**Comment by zeus6768** [comment]
# 문제 상황

## 1. 새로고침 시 로그아웃

### 시나리오
1. 로그인 성공
2. 브라우저 새로고침 (F5 또는 ⌘+r)

## 2. 템플릿 보기 페이지에서 닉네임 미표시

- 닉네임 대신 `코드잽` 표시

### 시나리오
1. 로그인 성공
2. '**코드잽**님의 템플릿 입니다 :)' 출력

## 3. 상세보기 페이지에서 파일명 리스트 출력 잘림

테스트 환경 해상도: 1680 * 1050

<img width="210" alt="Screenshot 2024-08-08 at 11 22 09 AM" src="https://github.com/user-attachments/assets/4dea5bc2-fc57-47c6-a1ca-8bec7a9f303c">



# 개선 제안

## 1. 브라우저에 비밀번호 저장 시 이메일 대신 사용자명 자동완성

### 시나리오
1. 회원가입 성공
2. 브라우저에 비밀번호 저장
4. 로그인 페이지 접속

### 해결방법
- username 때문인 것으로 보임

## 2. 탭키로 스니펫 간 이동

- 마우스를 써야 하는 게 불편해요. 
- 편집기에서 탭 키로 포커스를 빠져나갈 수 있으면 좋겠어요. 

## 3. 태그 선택으로 템플릿 상세조회 후 뒤로가기 하면 초기화 되지 않게 변경

현재 불편하다고 느끼는 시나리오는 다음과 같습니다. 

1. 로그인
2. 템플릿에서 태그 선택
3. 템플릿 상세보기
4. 뒤로가기
5. 전체보기

여기서 전체보기가 아니라, 2번에서 선택한 태그에 해당하는 템플릿이 보여야 합니다.

**Reply by vi-wolhwa** [reply]
문제상황 3. `SelectList`에서 파일명을 중략할 때 확장자를 생략하지 않도록 하는 방향으로 리팩토링 하겠습니다.

**Reply by Jaymyong66** [reply]
해결 - _2. 템플릿 보기 페이지에서 닉네임 미표시_

**Comment by kyum-q** [comment]
키워드 검색 시 스니펫 파일명, 코드에 포함되도 검색된다
여기서 ddd 템플릿 안에 111이라는 문자가 포함되어 있음 하지만 검색 시 스니펫은 보이지 않아서 검색 오류가 난 것 같다
> UI에서 어디서 포함됬는지 보여주면 더 좋을 것 같은데 다 상관없다면 나도 뭐~
![image](https://github.com/user-attachments/assets/3c030dcb-86ff-4d11-a57c-9e4403c12eb9)

![image](https://github.com/user-attachments/assets/4a94583e-82c1-4c48-9833-dafc6425cce2)

**Comment by kyum-q** [comment]
생성 시 이미 태그가 존재하는 경우 추가 안되게

> UI에서 추가는 되고 나중에 저장하고 확인하면 중복된 태그는 삭제되긴 함

### 생성 시
![image](https://github.com/user-attachments/assets/b8675134-6572-4ee2-b882-aa2f5da25e01)

### 생성 후 확인
![image](https://github.com/user-attachments/assets/1233e793-d8c0-4438-a426-167fd42aabad)

**Comment by HoeSeong123** [comment]
카테고리를 생성할 때 곧바로 선택되지 않음

## 원하는 시나리오
1. 카테고리를 생성하고 엔터를 누르면 해당 카테고리가 바로 선택된다.

## 현재 상황
1. 카테고리를 생성하고 엔터를 누르면 카테고리 추가는 되지만 여전히 새 카테고리 생성 칸이 된다.
<img width="231" alt="스크린샷 2024-08-08 오전 11 11 56" src="https://github.com/user-attachments/assets/798b6557-cca5-4396-b69b-cd6460c886dd">
-> 
<img width="326" alt="스크린샷 2024-08-08 오전 11 12 11" src="https://github.com/user-attachments/assets/82312279-6e54-4a2f-8ddb-9fb92118af9e">
2. 카테고리를 선택해야 선택된다.

**Comment by kyum-q** [comment]
(의견)
창 작아졌을 때 카테고리 접히는 브레이크 포인트가 너무 넓어요 ~

**Comment by kyum-q** [comment]
검색 시 정렬되는 것을 생성 시간이 아니라 수정 시간으로 해서 템플릿 옆에 보이는 시간하고 일치했으면 좋겠어요

**Reply by kyum-q** [reply]
상세 조회 시 생성 시간도 같이 보여주세요

**Reply by kyum-q** [reply]
목록 조회에서는 수정 시간을 보여주세요 (확정)
그리고 수정된 경우 [수정됨] 라는 식으로 표시가 있었으면 좋겠어요 (논의 필요)

**Reply by kyum-q** [reply]
드롭다운에 `최신순`, `오래된순` 으로 변경

**Reply by kyum-q** [reply]
월하가 sort하는 기준을 createAt을 modifedAt으로 변경해주세요

**Reply by kyum-q** [reply]
API response에 createdAt 추가해서 보내기

**Comment by zangsu** [comment]
화면 크기가 줄어들었을 때 스니펫 목록 중 선택 안된 스니펫이 더 잘 보였으면 좋겠어요

**Comment by Jaymyong66** [comment]
프론트 CD 파일 수정

**Comment by Jaymyong66** [comment]
- 로그인 실패 토스트 색상 변경
- 코드잽 로고 변경

**Comment by Jaymyong66** [comment]
Header에서 구경가기 탭 제거

**Comment by HoeSeong123** [comment]
uri를 통해 다른 사람의 템플릿 상세 조회에 접근하려고 하면 로그아웃이 됨.

**Comment by HoeSeong123** [comment]
생각나서 적어놓음
- 카테고리 삭제 기능
- 오류 시 alert창을 통해 사용자에게 알려주는 기능
- 카테고리의 종류가 너무 많은 경우에 대한 고민..?

**Comment by HoeSeong123** [comment]
<img width="1371" alt="스크린샷 2024-08-08 오후 2 57 37" src="https://github.com/user-attachments/assets/626dc4b1-0e77-4d13-9dc5-9c83fe5dea3e">

어떤 카테고리를 선택하고 그 상태에서 뒤로가기를 누르면 저런 alert창이 뜨던데 왜 그런건가요??

**Comment by Jaymyong66** [comment]
코드 템플릿 편집 / 업로드 시, 코드 스니펫이 너무 작다(높이, 10줄 정도?)

**Comment by kyum-q** [comment]
![image](https://github.com/user-attachments/assets/f7e67543-3a40-4dcf-9f87-e081e8a12c9e)

**Comment by kyum-q** [comment]
![image](https://github.com/user-attachments/assets/1756b006-d69e-4fa7-9ff1-90fe7aafa454)

-> 코드 창 크기가 고정 아니었으면 좋겠어요
